Taiga update

JostCrow opened the pull request.

ramezkabbani committed:
Add inspect to import, remove apps from import

qwertyllo committed:
Disallow anonymous users to see profile details

Currently anonymous users can see other user profiles, including personal data (name, surname).
This change restrict this permission so only auth users can see other user profiles.

qwertyllo committed:
Allow to configure whether anonymous users can see other users profile details

andreashaerter committed:
FRONT_SITEMAP_CACHE_TIMEOUT listed twice in example

FRONT_SITEMAP_CACHE_TIMEOUT was listed twice, should be
STATS_CACHE_TIMEOUT the second time. Propably a Copy & Paste
error.

Commit:
Emails redesigned

hecfernandez committed:
send notifications filtered by model type

Commit:
Custom signature and product_name in email templates

hecfernandez committed:
filter notifications according to typology

hecfernandez committed:
comments description fixes

hecfernandez committed:
add a feature toggle for custom notifications filter

Commit:
New email icons

Commit:
Fix footer links style on mail templates

Commit:
Improve email logos

hecfernandez committed:
fix notifications custom filter

hecfernandez committed:
adjust reduced filter for history entries

hecfernandez committed:
fix reduced filter for issues notifications

migonzalvar committed:
Initial version

Commit:
Promote task and issues to user story with watchers, attachments and comments

migonzalvar committed:
Update translation

migonzalvar committed:
fix: Object could no attribute

The called `project` could be other object as user story, issue...

migonzalvar committed:
Force order to remove pagination warning

migonzalvar committed:
Update change log

migonzalvar committed:
fix: Send a different email for each admin

It seems djmail does not support to pass a list of recipients to the
underlying Django send mail function.

migonzalvar committed:
Update change log

migonzalvar committed:
feat(i18n): Add Latvian translation

hecfernandez committed:
fix settings use in notifications tests

Commit:
Fix typo

hecfernandez committed:
sanitize user full name for profile update

hecfernandez committed:
sanitize user full name for register

Commit:
Updated translatable strings catalog

hecfernandez committed:
prevents links insertion in full name field

theriverman committed:
[issues/1391] fix: permalink serialization

migonzalvar committed:
Merge pull request #1387 from hecfernandez/reduce-notifications-upstream

Reduce notifications

migonzalvar committed:
feat(i18n) Update catalog for English

hecfernandez committed:
optimize tasks export query

Commit:
Fix support URL

migonzalvar committed:
fix: Check if slug attribute exists

Thank you @Meissnerim for pointing it out in
https://github.com/taigaio/taiga-back/pull/1397

migonzalvar committed:
Update change log

theriverman committed:
FIX for accidental over-indent

theriverman committed:
[issues/1233]: Fixed bug preventing admin action

A faulty import statement raised `AttributeError` in `taiga.projects.admin` when `anon_permissions` gets initialized during the execution of the `Make public`  admin action.

Additionally some style faults, such as, line breaks, white spaces and indents have been corrected.

theriverman committed:
typo fix in `external_apps`

hecfernandez committed:
remove projects filter for public user

hecfernandez committed:
protect get project stats method against role points without user story

hecfernandez committed:
remove print

migonzalvar committed:
Merge pull request #1402 from hecfernandez/fix-get-project-stats

Fix get project stats

migonzalvar committed:
(i18n) Update Basque language

migonzalvar committed:
(i18n) Update translation

migonzalvar committed:
(i18n) Update Korean translation

migonzalvar committed:
Update change log

migonzalvar committed:
Merge branch 'dangarbar-ft/custom-mail-templates'

migonzalvar committed:
ref: Update dependencies

CarlosLVar committed:
String review (#1405)

Updated several strings.

migonzalvar committed:
(i18n) Update EN catalog

migonzalvar committed:
ref(i18n): Clear translation for English

migonzalvar committed:
ref: Add explicit file close

Commit:
generate a history entry when promoting a task or issue (#1406)

migonzalvar committed:
fix: Get actual slug of obj different from project

It fixes wikilinks render on issues, tasks... ie no WiKi pages.

Fixes #1408

migonzalvar committed:
style: Fix EOL

migonzalvar committed:
feat: Update dependencies

migonzalvar committed:
ref: Update requirements

migonzalvar committed:
ref: Update requirements

Commit:
i18n(tx): Fetch latest translations

migonzalvar committed:
(i18n) Updated translation file

migonzalvar committed:
Update change log

alextreme commented:
@JostCrow we maken gebruik van de 'maykin' branch, daar ik enkele Taiga wijzigingen heb gemaakt. Master bijwerken mag als je dat wilt maar die gebruiken we dus niet.

timgates42 committed:
docs: Fix simple typo, accesible -> accessible

There is a small typo in tests/integration/test_users.py.

Should read `accessible` rather than `accesible`.

Commit:
Fix recovery of project due date values when importing

migonzalvar committed:
Use set to compare results to avoid sorting differences

Commit:
Fix assigned_users signal for web notifications

Commit:
Exclude non-member users from mentions

Commit:
Fix mentions tests

migonzalvar committed:
Migrate from pipenv to pip-tools

migonzalvar committed:
Migrate from pipenv to pip-tools (5/)

migonzalvar committed:
Migrate from pipenv to pip-tools (3/)

migonzalvar committed:
Migrate from pipenv to pip-tools (4/)

migonzalvar committed:
Migrate from pipenv to pip-tools (2/)

Commit:
Fetch locales

migonzalvar committed:
Update change log

migonzalvar committed:
Make field create only

Commit:
Add security contributor.

Commit:
Add security contributor

migonzalvar committed:
Update change log

migonzalvar committed:
Test only python3.7

migonzalvar committed:
Fail on every warning

migonzalvar committed:
Fix deprecation warning

migonzalvar committed:
Ignore RemovedInDjango20Warning in migrations

hecfernandez committed:
update django middleware implementation

hecfernandez committed:
update user authentication in throttling code

hecfernandez committed:
update django remote fields implementation

migonzalvar committed:
Do not raise error on warnings in travis

hecfernandez committed:
resolve several deprecation issues in base api code

hecfernandez committed:
resolve deprecatio warning with django virtual fields

hecfernandez committed:
resolve deprecation with django auth methods

hecfernandez committed:
update pytest warnings config

hecfernandez committed:
fix deprecation warning with django remote fields

hecfernandez committed:
fix deprecation warning with direct assignment in m2m relations

hecfernandez committed:
fix deprecation warning with direct assignment in timeline test

domenicolino committed:
ready-for-test status is currently implying a ticket is closed. With this small fix, at least for new installations this issue will be fixed.

hecfernandez committed:
resolve getfullargspec issue with return values

hecfernandez committed:
add support for CallableBool in json renderer

hecfernandez committed:
fix middleware callbacks

migonzalvar committed:
Activate python3.6 again

migonzalvar committed:
Set warnings to default

migonzalvar committed:
Set DeprecationWarning to default

migonzalvar committed:
Merge pull request #1433 from taigaio/remove-deprecation-warning

Solve RemovedInDjango20Warning

hecfernandez committed:
fix django url lib usage

migonzalvar committed:
Add scripts to install and update requirements in venv

hecfernandez committed:
fix foreing keys deprecation warning

hecfernandez committed:
fix URLPattern deprecation implementation

hecfernandez committed:
fix urls config

hecfernandez committed:
remove old deprecation warnings from pytest config

hecfernandez committed:
fix foreing key declaration errors in migrations

migonzalvar committed:
Merge pull request #1427 from taigaio/200430-pip-tools

Migrate from pipenv to pip-tools

migonzalvar committed:
Add exec permission

migonzalvar committed:
Upgrade dependencies

hecfernandez committed:
update requirements.txt

hecfernandez committed:
update requirements

hecfernandez committed:
update requirements

hecfernandez committed:
remove permissions meta option from models

hecfernandez committed:
fix post data in attachment test

migonzalvar committed:
Merge pull request #1109 from qwertyllo/patch-1

Disallow anonymous users to see profile details

Commit:
Update fr, lv, ru locales

migonzalvar committed:
Update change log

migonzalvar committed:
Add createcachetable

migonzalvar committed:
Merge pull request #1439 from hecfernandez/upgrade-django2.2

Upgrade django2.2

migonzalvar committed:
Upgrade requirements

migonzalvar committed:
Upgrade psycopg2

migonzalvar committed:
Fix missing import

migonzalvar committed:
Use django.url.path properly

migonzalvar committed:
Adapt delete_selected admin action to Django 2.2

migonzalvar committed:
Add missing migrations

migonzalvar committed:
Fix bug

This commit restrict response inspection to valid responses, those with
status code 20x.

migonzalvar committed:
[5.0.x] Fix bug

This commit restrict response inspection to valid responses, those with
status code 20x.

Backport of 1646e49587604e56a6152a37872fe6e4eb5639b0 from master.

migonzalvar committed:
Update change log

hecfernandez committed:
Fix import projects for epics with external related user stories (#1449)

hecfernandez committed:
[5.0.x] Fix import projects for epics with external related user stories (#1449)

Backport of 023a25d05cbe881e8c7be6c05f2acdb62628e908 from master.

migonzalvar committed:
Update change log

migonzalvar committed:
Merge branch 'stable'

migonzalvar committed:
[5.0.x] Fix import of legacy projects

Backport of eed7b0078058009e0cc9a4a4a84ceb53369ebb8b from master.

migonzalvar committed:
Fix import of legacy projects

migonzalvar committed:
Update change log

migonzalvar committed:
fix(import): Ensure owner is admin

When importing a project, the user should be owner and admin.

hecfernandez committed:
add attachament description data in promote process

hecfernandez committed:
exclude delete comments on task/issue promote to us

hecfernandez committed:
remove unnecessary code

migonzalvar committed:
Merge pull request #1459 from hecfernandez/fix-promote-export-data

Fix issue/task "promote to us" export data

hecfernandez committed:
Fix promote task export comments (#1460)

Keep deleted comments in task promote process

migonzalvar committed:
Keep user who deleted the comment

Related 908628dd263d660e3d3ff6b01be0b783b93906a5

hecfernandez committed:
fix field to native method

hecfernandez committed:
Task promotion creates user story and deletes original task (#1454)

See TG101.

migonzalvar committed:
Verify user email (#1461)

See TG459

migonzalvar committed:
Update requirements

migonzalvar committed:
(i18n) Add default option

migonzalvar committed:
(i18n) Regenerate .po catalog

migonzalvar committed:
(i18n) Update trasnalations

migonzalvar committed:
fix: Maintain verified_email field on email change

Commit:
Updates es locale

migonzalvar committed:
Update change log

migonzalvar committed:
Merge branch 'master' into stable

migonzalvar committed:
feat: Prevent member creation to users with unverified email address

bameda committed:
chore: create github action tests-and-coverall-app.yml

migonzalvar committed:
feat(admin) Filter by verified email status

migonzalvar committed:
(i18n) Update message catalogs

migonzalvar committed:
Update change log

migonzalvar committed:
Avoid cache too short texts

migonzalvar committed:
Add prefix to cache key

migonzalvar committed:
(mdrender) Set timeout to cache entry

migonzalvar committed:
Merge pull request #1470 from taigaio/ft/mdrender-cache

Improve mdrender cache handling

migonzalvar committed:
Update change log

carlosgoce committed:
Fix typo

carlosgoce committed:
Fix an HTTP 500 when dictionaries were sent as login credentials

carlosgoce committed:
Parametrize mdrender moving parameters to settings

migonzalvar committed:
Update change log

migonzalvar committed:
Upgrade requirements

migonzalvar committed:
fix(tests) Upgrade factory_boy 3

migonzalvar committed:
(i18n) Update fr locale

migonzalvar committed:
Update change log

CarlosLVar committed:
fix: Only shows verify button when unverified (#1480)

Co-authored-by: Carlos L. Var <carlos.letamendia@revelio.dev>

migonzalvar committed:
i18n: Update catalog

migonzalvar committed:
i18n: Update fa

migonzalvar committed:
i18n: Update all languages with new strings

migonzalvar committed:
feat: Unset verified_email for new invited user

An user created from an invitation needs to verify their email as it was
a regular public registered user.

migonzalvar committed:
i18n: Update translation

migonzalvar committed:
Update change log

migonzalvar committed:
Add security contributor

bameda committed:
chore(ci): remove TravisCI integration

migonzalvar committed:
Upgrade requirements

migonzalvar committed:
Remove direct dependency and unpin

migonzalvar committed:
Unpin and update

migonzalvar committed:
Use raw string to avoid DeprecationWarning

migonzalvar committed:
Allow upgrade last patch and update

migonzalvar committed:
chore: Add more Python versions

migonzalvar committed:
Update change log

Commit:
Corrects signup template html rendering

diacritica committed:
taiga6 notice

bameda committed:
chore(actions): fix some error

farooqaaa committed:
Fix misspelled variable name

farooqaaa committed:
Remove unused imports

farooqaaa committed:
Disable linter warnings for wildcard imports

Commit:
Add new security contributor

migonzalvar committed:
feat: Returns refresh fragment on attachments

It adds the refresh fragment to the attachment URL. It can be used to
identify the attachment object when included in other fields.

This commit also includes a refactor to move funciotns from
mdrender.extensions.refresh_attachment to the common
projects.attachments.services module.

migonzalvar committed:
Upgrade requirements

migonzalvar committed:
fix: Use the right factory

migonzalvar committed:
fix: Use model to build refresh token

In general content_type.name and content_type.model are equiavalent but not for wiki pages.

migonzalvar committed:
Upgrade requirements

Commit:
Update locales

migonzalvar committed:
Add initial version of Arabic

migonzalvar committed:
Update change log
